subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPluginExtension.java

```diff
@@ -132,29 +130,48 @@ public <T extends Task & JavaForkOptions> void applyTo(final T task) {
         );
     }
 
-    private static class JacocoOutputCleanupTestTaskAction implements Action<Task> {
+    private static class JacocoPrepareTestTaskAction implements Action<Task> {
```

Comment:
This class now does 2 things that are not obvious from the type name.  Can you please document this with javadoc on this type?